Automattic/pocket-casts-ios
# Referrals IAP async calls

| 📘 Part of: #2083  |  
|:---:|

Fixes # 

This updates the Referrals IAP fetch to be async so if for some reason the IAP info is not available it can be loaded

| Send | Claim |
| - | - |
| ![Simulator Screenshot - iPhone 16 Pro - 2024-10-10 at 16 37 39](https://github.com/user-attachments/assets/c9f0e345-141a-40a9-8dbf-98e3187def0a) | ![Simulator Screenshot - iPhone 16 Pro - 2024-10-10 at 16 34 02](https://github.com/user-attachments/assets/79e2d319-de77-4d6a-b1ae-9163e76edfcb) |

## To test

1. Start the app on a real device
2. Ensure that you the referrals FF enabled
3. Login with an account that has Plus or Patron
4. Go to Profile tab
5. Check that you see the Gift icon on the top left
6. Tap on the Gift icon
7. Check that the referral Info is show correctly ( 2-Month)
8. Tap on send pass and save the referral URL to the Reminders app
9. Log out and log in with a new account that never had a subscription
10. Switch to the Reminders app and tap on the referral link
11. The app should open with the claim screen
12. Check that the info regarding the referral duration ( 2 Month) and price after referral period ends is show correctly

## Checklist
